cypress/page/modal.page.ts

```diff
@@ -0,0 +1,34 @@
+class Modal{
+    private tableSelector: String;
+    constructor(){
+        this.tableSelector = ".modal-body tbody";
+    }
+    public checkingModal(personalInformation: any): void {
+        /* Coverts the json in a bidimensional array and selects only
+         the second column (the values) */
+        let data = Object.entries(personalInformation).map((item) => { 
+            return item[1]
+        });
+        /* Concats the name with the last name to get the full name
+        and deletes the element that contains just the last name */
+        data[0] = data[0].concat(" ", data[1]);
+        data.splice(1, 1);
+        /* Gets the hobbies (it's the only element that it's an array)
+        and sorts them and reverse (since the options are in reverse 
+        alphabetical order). Then, convert to string joining with a comma */
+        data[data.indexOf(data.find(item => item instanceof Array))] =
+            data[data.indexOf(data.find(item => item instanceof Array))].sort().reverse().join(", ");
+        
+        /* Concats the state and city to get the full address and deletes
+        the element that contains just the city */
+        data[data.length - 2] = data[data.length - 2].concat(" ", data[data.length - 1]);
+        data.pop();
+
+        data.forEach((item) => {
+            cy.get(`${this.tableSelector} 
+            tr:nth-child(${data.indexOf(item)+1}) td:nth-child(2)`).should("contain.text", item)
+        })
+    }
```

Comment:
you can use several strategies to simplify the verification:

1. The main goal is to verify that all the information that we filled in the form is present in the modal. So, you can use a should contain, the contain will find whichever you want:
    - `cy.get(this.tableSelector).should("contain", personalInformation.fieldToValidate)`
    - fieldToValidate can be any field that you want to search: name, date, hobbies, etc.
2. For the full name, you can use string interpolation: 
    ```js
    `${personalInformation.name} ${personalInformation.lastName}`
    ```
3. For the hobbies you can use the join method of the javascript array object: `personalInformation.hobbies.join(", ")`
4. The date, you can convert the date string into a [Date object](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date). And you can take advantage of some functions of that object to convert "27 Jul 2016" to "27 July,2016".Look:
    ```js
    const date = new Date(personalInfo.dateOfBirth);
    const longMonth = date.toLocaleString("default", {
      month: "long",
    });
    const expectedDate = `${date.getDate()} ${longMonth},${date.getFullYear()}`
    ```